cmd/erasure-object.go

```diff
@@ -1029,30 +1035,42 @@ func (er erasureObjects) putObject(ctx context.Context, bucket string, object st
 		parityOrig := parityDrives
 
 		atomicParityDrives := uatomic.NewInt64(0)
+		atomicOfflineDrives := uatomic.NewInt64(0)
+
 		// Start with current parityDrives
 		atomicParityDrives.Store(int64(parityDrives))
 
 		var wg sync.WaitGroup
 		for _, disk := range storageDisks {
 			if disk == nil {
 				atomicParityDrives.Inc()
+				atomicOfflineDrives.Inc()
 				continue
 			}
 			if !disk.IsOnline() {
 				atomicParityDrives.Inc()
+				atomicOfflineDrives.Inc()
 				continue
 			}
 			wg.Add(1)
 			go func(disk StorageAPI) {
 				defer wg.Done()
 				di, err := disk.DiskInfo(ctx)
 				if err != nil || di.ID == "" {
+					atomicOfflineDrives.Inc()
 					atomicParityDrives.Inc()
 				}
 			}(disk)
 		}
 		wg.Wait()
 
+		if int(atomicOfflineDrives.Load()) > len(storageDisks)/2 {
```

Comment:
Wouldn't this catch more cases?
```suggestion
		if int(atomicOfflineDrives.Load()) >= len(storageDisks+1)/2 {
```

Reply:
It can actually, let me add this suggestion.